mypy/errors.py

```diff
@@ -11,6 +11,11 @@
 from mypy.version import __version__ as mypy_version
 from mypy.errorcodes import ErrorCode
 from mypy import errorcodes as codes
+from mypy.util import (
+    decode_python_encoding, DecodeError, trim_source_line, DEFAULT_SOURCE_OFFSET,
+    WIGGLY_LINE, DEFAULT_COLUMNS, MINIMUM_WIDTH
+)
+from mypy.fscache import FileSystemCache
```

Comment:
I think it would be nice if there was no import from `mypy.fscache` in this file, and instead we'd accept a callback that reads file contents, for example (dependency injection). We could probably also get rid of the `decode_python_encoding` and `DecodeError` imports.

Reply:
OK, I was thinking about a protocol, but callback is even better.